api/services/userService.js

```diff
@@ -0,0 +1,37 @@
+const jwt = require("jsonwebtoken");
+const axios = require("axios");
+
+const { userDao } = require("../models");
+
+const kakaoSignUp = async (kakaoToken) => {
+  const getKakaoUserData = await axios.get(
+    "https://kapi.kakao.com/v2/user/me",
+    {
+      headers: {
+        Authorization: `Bearer ${kakaoToken}`,
+      },
+      "Content-type": "application/x-www-form-urlencoded;charset=utf-8",
+    }
+  );
+
+  const kakaoId = getKakaoUserData.data.id;
+  const profileNickname = getKakaoUserData.data.properties.nickname;
+  const accountEmail = getKakaoUserData.data.kakao_account.email;
+  const ageRange = getKakaoUserData.data.kakao_account.age_range;
+
+  const ifCheckKakaoId = await userDao.checkKakaoId(kakaoId);
+
+  if (!ifCheckKakaoId) {
+    await userDao.kakakoSignup(
+      kakaoId,
+      profileNickname,
+      accountEmail,
+      ageRange
+    );
+  }
+  const accessToken = await jwt.sign({ id: kakaoId }, process.env.JWT_SECRET);
```

Comment:
```suggestion
  const accessToken = await jwt.sign({ id: user.id }, process.env.JWT_SECRET);
```
카카오에서 발급하는 id가 아닌, 우리 database의 pk값을 통해 토큰을 만들 수 있도록 해주세요. 이에 맞춰 if문으로 분기되는 로직도 적절히 수정되어야겠습니다. "토큰은 결국 누구나 쉽게 payload에 담기는 data를 볼 수 있다"라고 했습니다. 카카오에서 주는 user info는 올바른 인증과정을 거쳐, 인가코드와 카카오 토큰을 통해 받은 데이터이며 외부 api를 통해 받은 해당 플랫폼의 user info입니다. 이에 대한 데이터는 보안상 문제가 생기지 않도록 해당 플랫폼들에서도 언급을 하고 있습니다. 

우리가 관리하는 DB의 pk값을 활용할 수 있도록 해주세요.